File: app/src/main/java/com/jiangkang/ktools/requests/zhihu/ZhihuFragment.java
package com.jiangkang.ktools.requests.zhihu;


import android.os.Bundle;
import android.support.annotation.MainThread;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.gson.Gson;
import com.jiangkang.ktools.R;
import com.jiangkang.requests.KRequests;
import com.jiangkang.requests.zhihu.ZhihuApi;
import com.jiangkang.requests.zhihu.bean.LatestNews;
import com.jiangkang.tools.utils.LogUtils;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;

/**
 * A simple {@link Fragment} subclass.
 */
public class ZhihuFragment extends Fragment {


    public ZhihuFragment() {

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_zhihu, container, false);
    }


    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        loadData();
    }

    private void loadData() {
        getLatestDailyNews();
    }

    private void getLatestDailyNews() {
        KRequests.request(ZhihuApi.class)
                .getLatestNews()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Consumer<LatestNews>() {
                    @Override
                    public void accept(LatestNews latestNews) throws Exception {
                        LogUtils.json(new Gson().toJson(latestNews));
                    }
                });
    }
}

File: app/src/main/java/com/jiangkang/ktools/requests/zhihu/ZhihuFragment.kt
package com.jiangkang.ktools.requests.zhihu


import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import com.google.gson.Gson
import com.jiangkang.ktools.R
import com.jiangkang.requests.KRequests
import com.jiangkang.requests.zhihu.ZhihuApi
import com.jiangkang.requests.zhihu.bean.Story
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.fragment_zhihu.*


/**
 * A simple [Fragment] subclass.
 */
class ZhihuFragment : Fragment() {

    lateinit var adapter: ZhiHuAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activity.title = "知乎"
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater!!.inflate(R.layout.fragment_zhihu, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        rcZhiHu.layoutManager = LinearLayoutManager(context, LinearLayout.VERTICAL, false)

        val viewModel = ViewModelProviders.of(this).get(ZhihuListViewModel::class.java)

        observeViewModel(viewModel)

//        loadData()
    }

    private fun observeViewModel(viewModle: ZhihuListViewModel) {

        viewModle.storyListObservable.observe(this, Observer<ArrayList<Story>> {
            adapter = ZhiHuAdapter(it!!)
            rcZhiHu.adapter = adapter
        })
    }

    private fun loadData() {
        getLatestDailyNews()
    }

    private fun getLatestDailyNews() {
        KRequests.request(ZhihuApi::class.java)
                .latestNews
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe { latestNews ->
                    run {
                        Log.d("zhihu", Gson().toJson(latestNews))
                        adapter = ZhiHuAdapter(latestNews.stories)
                        rcZhiHu.adapter = adapter
                    }
                }
    }


}
